src/main/java/graphql/nadel/Parser.java

```diff
@@ -22,8 +25,9 @@ public StitchingDsl parseDSL(String input) {
         CommonTokenStream tokens = new CommonTokenStream(lexer);
         StitchingDSLParser parser = new StitchingDSLParser(tokens);
         parser.removeErrorListeners();
+        parser.addErrorListener(ThrowingErrorListener.INSTANCE);
         parser.getInterpreter().setPredictionMode(PredictionMode.SLL);
-        parser.setErrorHandler(new BailErrorStrategy());
+        parser.setErrorHandler(new DefaultErrorStrategy());
```

Comment:
this will allow to pass through exception from ThrowingErrorListener, which generates detailed info about what went wrong.